hazelcast/include/hazelcast/client/HazelcastClient.h

```diff
@@ -705,17 +702,27 @@ namespace hazelcast {
             void shutdown();
 
         private:
+            std::auto_ptr<internal::nearcache::NearCacheManager> createNearCacheManager();
+
+            template <typename FACTORY>
+            boost::shared_ptr<spi::ClientProxy> getDistributedObjectForService(
+                    const std::string &serviceName, const std::string &name, FACTORY &factory) {
+                return proxyManager.getOrCreateProxy<FACTORY>(serviceName, name, factory);
+            }
+
             ClientConfig clientConfig;
             ClientProperties clientProperties;
             spi::ClientContext clientContext;
             spi::LifecycleService lifecycleService;
             serialization::pimpl::SerializationService serializationService;
             connection::ConnectionManager connectionManager;
+            std::auto_ptr<internal::nearcache::NearCacheManager> nearCacheManager;
```

Comment:
At java side, it uses a method to create it: createNearCacheManager which returns a pointer which uses a client extension which we do not have a cpp client. But i preferred using the same code as java does. Provided that its constructor parameters are already present, it can be done both ways. if createNearCacheManager does more complex things in the future, it may be better to keep it as a pointer.